tests/backup/backup_resiliency_test.go

```diff
@@ -1558,3 +1558,211 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Taking backup %s of application- %s", backupName, namespace))
+					}(backupName, namespace)
+				}
+				backupNamesMap[namespace] = backupNames
+			}
+			wg.Wait()
+			log.Infof("The list of backups taken are: %v", backupNames)
+		})
+
+		Step("Restoring the backed up applications", func() {
+			log.InfoD("Restoring the backed up applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range appNamespaces {
+				namespaceMapping := make(map[string]string)
+				for _, backupName := range backupNamesMap[namespace] {
+					restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+					restoreNames = append(restoreNames, restoreName)
+					time.Sleep(1 * time.Second)
+					namespaceMapping[namespace] = fmt.Sprintf("new-namespace-%v", time.Now().Unix())
+					wg.Add(1)
+					go func(restoreName string, backupName string, namespaceMapping map[string]string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+					}(restoreName, backupName, namespaceMapping)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of restores taken are: %v", restoreNames)
+		})
+
+		Step("Cancelling the ongoing restores", func() {
+			log.InfoD("Cancelling the ongoing restores")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, restoreName := range restoreNames {
+				wg.Add(1)
+				go func(restoreName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					restoreInspectRequest := &api.RestoreInspectRequest{
+						Name:  restoreName,
+						OrgId: orgID,
+					}
+					statusExpected := api.RestoreInfo_StatusInfo_InProgress
+					restoreProgressCheckFunc := func() (interface{}, bool, error) {
+						resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+						if err != nil {
+							return "", false, err
+						}
+						actual := resp.GetRestore().GetStatus().Status
+						if actual == statusExpected {
+							return "", false, nil
+						}
+						return "", true, fmt.Errorf("restore status for [%s] expected was [%v] but got [%s]", restoreName, statusExpected, actual)
+					}
+					_, err := task.DoRetryWithTimeout(restoreProgressCheckFunc, maxWaitPeriodForRestoreCompletionInMinute*time.Minute, restoreJobProgressRetryTime*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verfiying restore %s is in progress", restoreName))
+					err = DeleteRestore(restoreName, orgID, ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting restore %s while restore is in progress", restoreName))
+				}(restoreName)
+			}
+			wg.Wait()
```

Comment:
Over here, you're waiting for all restores to be created before youstart checking for inprogress state.
This could result in an error, where one of the restores completes _before_ all the restores finish starting up, which results a failure in the step where you check for inprogress, as inprogress will never happen.

A solution is to combine the two steps into one. In each thread, you can start the restore, check for inprogress and delete, and _then_ return from the thread. That way, the "restore-inprogress-delete" process is **independent** of other restores for any restore. 

Like:
```
for (...){
time.sleep
go func() {
start restore()
if inprogrss{
delete restore()
}
}
}
```

Reply:
If you see the restores are created without success check and will be triggered parallely, And coming to cancellation of restore's all will be triggered parallely without any dependent wait time or restore. 
As per testcase both should be seperate workflow.

Reply:
Thanks for the suggestion, I've solved this case by creating custom namespce with random string instead of time stamp and sleep.

Reply:
Alright.